packages/base/adapters/evm/ethers5/client.ts

```diff
@@ -91,37 +93,43 @@ export class Web3Modal extends Web3ModalScaffold {
 
   private walletConnectProviderInitPromise?: Promise<void>
 
-  private projectId: string
+  private projectId = ''
+
+  private options: AppKitOptions | undefined = undefined
 
   private chains: Chain[]
 
-  private chain: AvailableChain = CommonConstantsUtil.CHAIN.EVM
+  public chain: AvailableChain = CommonConstantsUtil.CHAIN.EVM
 
   private metadata?: Metadata
 
-  private options: Web3ModalClientOptions | undefined = undefined
+  public networkControllerClient: NetworkControllerClient
+
+  public connectionControllerClient: ConnectionControllerClient
+
+  public siweControllerClient = this.options?.siweConfig
+
+  public defaultChain: CaipNetwork | undefined = undefined
 
-  public constructor(options: Web3ModalClientOptions) {
-    const {
-      ethersConfig,
-      siweConfig,
-      chains,
-      defaultChain,
-      tokens,
-      chainImages,
-      _sdkVersion,
-      ...w3mOptions
-    } = options
+  public tokens = HelpersUtil.getCaipTokens(this.options?.tokens)
+
+  public constructor(options: AdapterOptions) {
+    const { ethersConfig, siweConfig, chains, defaultChain, tokens } = options
 
     if (!ethersConfig) {
       throw new Error('web3modal:constructor - ethersConfig is undefined')
     }
 
-    if (!w3mOptions.projectId) {
-      throw new Error('web3modal:constructor - projectId is undefined')
-    }
+    this.ethersConfig = ethersConfig
```

Comment:
Since we are passing the `ethersConfig` to the `EVMEthers5Adapter` class, it's better to keep here. But once we refactor our interface a bit and if we want to pass this to `createAppKit` or if we want to get rid of the `ethersConfig` object directly, we could do that in the related PR. For not, it's better place to have